eva_submission/retrieve_eload_and_project_from_lts.py

```diff
@@ -0,0 +1,135 @@
+                except:
+                    logging.warning(f'No Project accession found in ELOAD config for ELOAD {archive_name}')
+        else:
+            logger.warning(f"No ELOAD config file found for eload: {archive_name}")
+
+    def update_path_in_eload_config(self, retrieved_dir, archive_name):
+        config_file_path = os.path.join(retrieved_dir, archive_name, f'.{archive_name}_config.yml')
+
+        if os.path.isfile(config_file_path):
+            with fileinput.input(config_file_path, inplace=True) as file:
+                for line in file:
+                    new_line = line.replace('/nfs/production3/', '/nfs/production/keane/')
+                    print(new_line, end='')
+
+    def retrieve_archive(self, archive_path, retrieval_output_path, files_dirs_to_retrieve=''):
+        if os.path.exists(archive_path):
+            command = f"tar -xf {archive_path} -C {retrieval_output_path} {files_dirs_to_retrieve}"
+            command_utils.run_command_with_output('Retrieve files/dir from tar', command)
+        else:
+            logger.error(f'Archive path {archive_path} does not exist')
+
+    def get_files_or_dirs_to_retrieve_from_archive(self, archive):
+        files_dirs_to_retrieve = []
+        if 'retrieve_dirs' in archive and archive['retrieve_dirs']:
+            files_dirs_to_retrieve.extend(archive['retrieve_dirs'])
+        if 'retrieve_files' in archive and archive['retrieve_files']:
+            files_dirs_to_retrieve.extend(archive['retrieve_files'])
+
+        return ' '.join(files_dirs_to_retrieve)
+
+    def retrieve_eload(self, eload, retrieve_associated_project, update_path, eload_dirs_files, eload_lts_dir,
+                       project_lts_dir, eload_output_dir, project_output_dir):
+        self.create_dir_if_not_exist(eload_output_dir)
+        logging.info(f"Retrieving Eloads")
+
+        # Retrieve eload
+        eload_tar = f'{eload}.tar'
+        eload_archive_path = os.path.join(eload_lts_dir, eload_tar)
+        retrieva_dirs_files = " ".join(eload_dirs_files) if eload_dirs_files and len(eload_dirs_files) > 0 else ''
+        self.retrieve_archive(eload_archive_path, eload_output_dir, retrieva_dirs_files)
+
+        # Uncompress files
+        files_to_uncompress = self.get_compressed_files_in_dirs(os.path.join(eload_output_dir, eload))
+        self.uncompress_files(files_to_uncompress)
+
+        # Retrieve associated project if specified
+        if retrieve_associated_project:
+            project_acc = self.get_project_from_eload_config(eload_output_dir, eload)
+            if project_acc:
+                self.create_dir_if_not_exist(project_output_dir)
+                self.retrieve_archive(os.path.join(project_lts_dir, f'{project_acc}.tar'), project_output_dir)
+                project_files_to_uncompress = self.get_compressed_files_in_dirs(
+                    os.path.join(project_output_dir, project_acc))
+                self.uncompress_files(project_files_to_uncompress)
```

Comment:
This is similar to `retrieve_project` so we should refactor.

Reply:
done